src/pages/Dashboard/Modals/SupplyWithdraw/index.tsx

```diff
@@ -14,37 +14,52 @@ import {
 import { IAmountFormProps } from 'containers/AmountForm';
 import { AuthContext } from 'context/AuthContext';
 import useUserMarketInfo from 'hooks/useUserMarketInfo';
+import useWithdraw, { UseWithdrawParams } from 'hooks/useWithdraw';
+import useSupply, { UseSupplyParams } from 'hooks/useSupply';
 import { useTranslation } from 'translation';
 import { Asset, TokenId } from 'types';
-import { formatApy } from 'utilities/common';
+import { formatApy, getBigNumber } from 'utilities/common';
 import SupplyWithdrawForm from './SupplyWithdrawForm';
 import { useStyles } from '../styles';
 
 export interface ISupplyWithdrawUiProps {
   className?: string;
   onClose: IModalProps['handleClose'];
-  asset: Asset | undefined;
+  asset: Asset;
+}
+
+export interface ISupplyWithdrawProps {
   userTotalBorrowBalance: BigNumber;
   userTotalBorrowLimit: BigNumber;
   dailyEarnings: BigNumber;
+  supply: (params: UseSupplyParams) => void;
+  redeem: (params: UseWithdrawParams) => void;
+  redeemUnderlying: (params: UseWithdrawParams) => void;
+  isTransactionLoading: boolean;
+  vTokenBalance: string | undefined;
 }
 
 /**
  * The fade effect on this component results in that it is still rendered after the asset has been set to undefined
  * when closing the modal.
  */
-export const SupplyWithdrawUi: React.FC<ISupplyWithdrawUiProps> = ({
+export const SupplyWithdrawUi: React.FC<ISupplyWithdrawUiProps & ISupplyWithdrawProps> = ({
   className,
   onClose,
   asset,
   userTotalBorrowBalance,
   userTotalBorrowLimit,
   dailyEarnings,
+  supply,
+  redeem,
+  redeemUnderlying,
+  isTransactionLoading,
```

Comment:
This should be separated into two loading states. We don't need to block users from making another transaction (say supply tokens) while one is loading (say borrow tokens)